server/jamovi/server/rowtracker.py

```diff
@@ -0,0 +1,68 @@
+
+class RowTracker:
+
+    def __init__(self):
+        self._removed_rows = []
+        self.state_id = 0
+
+    @property
+    def removed_row_ranges(self):
+        return self._removed_rows
+
+    @removed_row_ranges.setter
+    def removed_row_ranges(self, ranges):
+        self._removed_rows = ranges
+
+    @property
+    def is_edited(self):
+        return len(self._removed_rows) > 0
+
+    def log_rows_added(index, count):
+        changed = False
+        for removed_range in self._removed_rows:
+            if index < removed_range['index']:
+                removed_range['index'] += count
+                changed = True
+            else:
+                break
+
+        if changed:
+            self.state_id += 1
+
+    def log_rows_removed(self, start, end):
+        print('AA')
```

Comment:
print statement

Reply:
yeah sorry, i'm still working on this branch, i thought i'd put all the row tracking stuff in undo/redo as well because it's kind of part of it. I'm nearly finished it. Just looking into the status bar stuff